File: src/java/com/uts/modelo/producto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.uts.modelo;

/**
 *
 * @author TechUTS
 */
public class producto {
    private int id;
    private String codigoProducto;
    private String nombre;
    private String descripcion;
    private double precio;
    private int cantidad;
    private byte[] imagen; // Nuevo atributo imagen
    private String estado;

    public producto() {
    }

    public producto(int id, String codigoProducto, String nombre, String descripcion, double precio, int cantidad, byte[] imagen, String estado) {
        this.id = id;
        this.codigoProducto = codigoProducto;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.cantidad = cantidad;
        this.imagen = imagen;
        this.estado = estado;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCodigoProducto() {
        return codigoProducto;
    }

    public void setCodigoProducto(String codigoProducto) {
        this.codigoProducto = codigoProducto;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public byte[] getImagen() {
        return imagen;
    }

    public void setImagen(byte[] imagen) {
        this.imagen = imagen;
    }

    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }
    
    
}

File: web/js/producto.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

function llenarFormulario(fila){
    var id = $(fila).find(".id").text();
    var codigoProducto = $(fila).find(".codigoProducto").text();
    var nombre = (fila).find(".nombre").text();
    var descripcion = (fila).find(".descripcion").text();
    var precio = (fila).find(".precio").text();
    var cantidad = (fila).find(".cantidad").text();
    var estado = (fila).find(".estado").text();
    var imagenSrc = $(fila).find(".imagen img").attr("src");
    
    $("#txtId").val(id);
    $("#txtCodigoProducto").val(codigoProducto);
    $("#txtNombre").val(nombre);
    $("#txtDescripcion").val(descripcion);
    $("#txtPrecio").val(precio);
    $("#txtCantidad").val(cantidad);
    $("#txtEstado").val(estado);
    $("#imgPreview").attr("src", imagenSrc);
    $("#txtId option[selected]").removeAttr('selected');
    
}

$(document).ready(function(){
    $("#exampleModal").on("hidden.bs.modal", function(){
        $('form')[0].reset();
        $("#imgPreview").attr("src", "");  // Clear image preview
    });
   $(document).on('click', '.btnEditar', function(){
        llenarFormulario($(this).closest('tr'));
   });
   $(document).on('click', '.btnEliminar', function(){
        llenarFormulario($(this).closest('tr'));
   });
});